test/applications_integration_test.go

```diff
@@ -0,0 +1,294 @@
+package test
+
+import (
+	"fmt"
+	"io/ioutil"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/cucumber/godog"
+
+	"github.com/algorand/go-algorand-sdk/client/v2/algod"
+	"github.com/algorand/go-algorand-sdk/crypto"
+	"github.com/algorand/go-algorand-sdk/future"
+	"github.com/algorand/go-algorand-sdk/transaction"
+	"github.com/algorand/go-algorand-sdk/types"
+)
+
+var client *algod.Client
+var tx types.Transaction
+var transientAccount crypto.Account
+var applicationId uint64
+
+func anAlgodVClientConnectedToPortWithToken(v int, host string, port int, token string) error {
+	var err error
+	portAsString := strconv.Itoa(port)
+	fullHost := "http://" + host + ":" + portAsString
+	client, err = algod.MakeClient(fullHost, token)
+	gh = []byte("MLWBXKMRJ5W3USARAFOHPQJAF4DN6KY3ZJVPIXKODKNN5ZXSZ2DQ")
+
+	return err
+}
+
+func iCreateANewTransientAccountAndFundItWithMicroalgos(microalgos int) error {
+	var err error
+
+	transientAccount = crypto.GenerateAccount()
+
+	status, err := acl.Status()
+	if err != nil {
+		return err
+	}
+	llr := status.LastRound
+
+	lv, err := acl.Versions()
+	gen = lv.GenesisID
+	gh = lv.GenesisHash
+	if err != nil {
+		return err
+	}
+
+	paramsToUse := types.SuggestedParams{
+		Fee:             types.MicroAlgos(fee),
+		GenesisID:       gen,
+		GenesisHash:     gh,
+		FirstRoundValid: types.Round(llr),
+		LastRoundValid:  types.Round(llr + 10),
+		FlatFee:         false,
+	}
+	ltxn, err := future.MakePaymentTxn(accounts[1], transientAccount.Address.String(), uint64(microalgos), note, close, paramsToUse)
+	if err != nil {
+		return err
+	}
+	lsk, err := kcl.ExportKey(handle, walletPswd, accounts[1])
+	if err != nil {
+		return err
+	}
+	ltxid, lstx, err := crypto.SignTransaction(lsk.PrivateKey, ltxn)
+	if err != nil {
+		return err
+	}
+	_, err = acl.SendRawTransaction(lstx)
+	if err != nil {
+		return err
+	}
+	err = waitForTransaction(ltxid)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func waitForTransaction(transactionId string) error {
+	status, err := acl.Status()
+	if err != nil {
+		return err
+	}
+	stopRound := status.LastRound + 10
+
+	for {
+		lstatus, err := acl.PendingTransactionInformation(transactionId)
+		if err != nil {
+			return err
+		}
+		if lstatus.ConfirmedRound > 0 {
+			break // end the waiting
+		}
+		status, err := acl.Status()
+		if err != nil {
+			return err
+		}
+		if status.LastRound > stopRound { // last round sent tx is valid
+			return fmt.Errorf("Transaction not accepted.")
+		}
+		time.Sleep(100 * time.Millisecond)
+	}
+	return nil
+}
+
+func iBuildAnApplicationTransactionWithTheTransientAccountTheCurrentApplicationSuggestedParamsOperationApprovalprogramClearprogramGlobalbytesGlobalintsLocalbytesLocalintsAppargsForeignappsAppaccounts(
```

Comment:
is the name a joke?)

Reply:
No. Cucumber is.

Reply:
Look at this: https://github.com/algorand/go-algorand-sdk/pull/169/files#diff-5390a1ea0a4c329071050d5e6d1b1092R284
you do not need to follow these weird suggestions from godog.